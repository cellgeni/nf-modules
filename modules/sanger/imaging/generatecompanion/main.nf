process IMAGINGGENERATECOMPANION {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "quay.io/cellgeni/imagetileprocessor:0.1.13"

    input:
    tuple val(meta), path(file_with_ome_md)

    output:
    tuple val(meta), path("${out_name}"), emit: csv
    tuple val(meta), path("${out_companion}"), emit: companion
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    out_name = "${prefix}_image_tiles.csv"
    out_companion = "${prefix}_image_companion.xml"
    """
    generate_companion.py run \\
        --file_with_ome_md ${file_with_ome_md} \\
        --tiles_csv ${out_name} \\
        --companion_xml ${out_companion} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imagepreprocess: \$(generate_companion.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    out_name = "${prefix}_image_tiles.csv"
    out_companion = "${prefix}_image_companion.xml"
    """
    touch ${out_name}
    touch ${out_companion}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imagepreprocess: \$(generate_companion.py version)
    END_VERSIONS
    """
}
