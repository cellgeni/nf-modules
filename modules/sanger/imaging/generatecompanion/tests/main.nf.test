// nf-test test main.nf.test in tests folder
nextflow_process {

    name "Test Process IMAGING_GENERATECOMPANION"
    script "../main.nf"
    process "IMAGING_GENERATECOMPANION"
    config "./nextflow.config"

    tag "modules"
    tag "modules_sanger"
    tag "imaging"
    tag "imaging/generatecompanion"

    test("generatecompanion - hcs") {

        when {
            params {
                module_args = "--master_file Index.xml"
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/TL_STP/TL_STP_RVT-HFF-20220623__2022-06-24T13_50_10-Measurement 1/Images', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("generatecompanion - wsi") {

        when {
            params {
                module_args = "--master_file Index.idx.xml"
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/QC/beadplatetest__2021-02-17T16_03_59-Measurement 1/Images', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("generatecompanion -  stub") {

        options "-stub"

        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + '/TL_STP/TL_STP_RVT-HFF-20220623__2022-06-24T13_50_10-Measurement 1/Images', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
