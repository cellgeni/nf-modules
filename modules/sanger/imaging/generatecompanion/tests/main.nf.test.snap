{
    "generatecompanion - hcs": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,b141a86e55c98ef279a69a760897d5ab"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,e85270402442371cdbf0efd9524d6fd1"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,e85270402442371cdbf0efd9524d6fd1"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,b141a86e55c98ef279a69a760897d5ab"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T22:31:28.934566029"
    },
    "generatecompanion - wsi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,60048616c857e95dc906b863187e5aff"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,689ed60d4c30d1b4eeecb8d714231594"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,689ed60d4c30d1b4eeecb8d714231594"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,60048616c857e95dc906b863187e5aff"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T22:28:57.572851169"
    },
    "generatecompanion - hcs - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_companion.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ],
                "companion": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_companion.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T21:48:28.374592138"
    },
    "generatecompanion -  stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_companion.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ],
                "companion": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_companion.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T22:17:58.079748614"
    }
}